### PR TITLE
Added missing dimension check for cost_y_ref

### PR DESCRIPTION
**Pull request by DennisWeir**

### CODE REVIEW

File: interfaces/acados_matlab_octave/acados_ocp.m [hunk 1]
```diff
@@ -133,9 +133,13 @@
...
             end
 
-            % create C object
-            obj.C_ocp = ocp_create(obj.model_struct, obj.opts_struct);
-
+            try
+                % create C object
+                obj.C_ocp = ocp_create(obj.model_struct, obj.opts_struct);
+            catch ex
+                sprintf('Exception:\n\t%s\n\t%s\n',ex.identifier,ex.message);
```
Comment (FreyJo) [inline interfaces/acados_matlab_octave/acados_ocp.m:140]: I like the `try`, probably it should return an error on failure, like this:
```
str = sprintf('Exception:\n\t%s\n\t%s\n',ex.identifier,ex.message);
error(str);
```

File: interfaces/acados_matlab_octave/ocp_compile_interface.m [hunk 2]
```diff
@@ -106,7 +106,7 @@ function ocp_compile_interface(opts)
 if ismac()
     FLAGS = 'CFLAGS=$CFLAGS -std=c99';
 else
-    FLAGS = 'CFLAGS=$CFLAGS -std=c99 -fopenmp';
+    FLAGS = 'CFLAGS=$CFLAGS -std=c99 -fopenmp -g'; % TODO debug mode
```
Comment (FreyJo) [inline interfaces/acados_matlab_octave/ocp_compile_interface.m:109]: why TODO?
Do you want to add the option of compiling in debug mode?

**Comment by DennisWeir** [comment]
Hi Jonathan,
when I created the pull request I had only changed ocp_create.c, it seems the system always takes the current branch version no matter at what commit the request was created. Sorry about that, I haven't used git extensively so far..
The other changes I made were to debug some issues I had with compilation. Therefor try/catch and the -g option, the todo was a note to myself to later remove the -g option in order to allow optimisation again. It turned out I didn't pass all dimensions correctly, causing access violations and Matlab to quit straight away. As soon as the dimensions were fixed everything ran smoothly!

**Comment by FreyJo** [comment]
No worries!

**Comment by DennisWeir** [comment]
Sure that's fine! I'll create a new branch next time :-)

**Comment by FreyJo** [comment]
Done, see #616

**Pull request closed without merging**